src/query/storages/fuse/src/operations/common/mutation_log.rs

```diff
@@ -97,6 +101,88 @@ pub struct CommitMeta {
     pub abort_operation: AbortOperation,
 }
 
+impl CommitMeta {
+    pub fn empty() -> Self {
+        CommitMeta {
+            conflict_resolve_context: ConflictResolveContext::ModifiedSegmentExistsInLatest(
+                SnapshotChanges::default(),
+            ),
+            abort_operation: AbortOperation::default(),
+        }
+    }
+}
+
+fn merge_conflict_resolve_context(
+    l: ConflictResolveContext,
+    r: ConflictResolveContext,
+    default_cluster_key_id: Option<u32>,
+) -> ConflictResolveContext {
+    match (l, r) {
+        (
+            ConflictResolveContext::ModifiedSegmentExistsInLatest(l),
+            ConflictResolveContext::ModifiedSegmentExistsInLatest(r),
+        ) => ConflictResolveContext::ModifiedSegmentExistsInLatest(SnapshotChanges {
+            removed_segment_indexes: l
+                .removed_segment_indexes
+                .into_iter()
+                .chain(r.removed_segment_indexes.into_iter())
+                .collect(),
+            added_segments: l
```

Comment:
A partial `TableMutationAggregator` cannot generate new segments by mutation_logs. Because the partition is in blocks, not in segments. So is `TableMutationAggregator` distributed necessary?